packages/fast-form-generator-react/src/form/form.tsx

```diff
@@ -216,7 +216,7 @@ class Form extends React.Component<IFormProps & IManagedClasses<IFormClassNameCo
         return <FormSection {...sectionProps} />;
     }
 
-    private handleOnChange = (location: string, data: any, isArray: boolean, index: number): void => {
+    private handleOnChange = (location: string, data: any, isArray: boolean, index: number, isChildren?: boolean): void => {
```

Comment:
The only way to know is by looking at the schema, which would get very complicated because of children nested (the same issue happened with arrays), I would need to pass in a sub-data location instead of the current location and pass up the schema and then look up the JSON schema type, so there would be the exact same number of arguments passed.